### ISSUE TITLE
Expose image alt text

### ISSUE THREAD

**Issue by AmeliaBR** [Shopify/buy-button-js#369]
The image templates should check for alt text & include it if available. 

If it is currently exposed in the data object, I couldn't figure out the variable names.

**Comment by robskrob** [comment]
@AmeliaBR did you ever find a solution to this?

**Comment by AmeliaBR** [comment]
@robskrob No, I didn't. Now that I check, the alt text is definitely there in the GraphQL responses that are coming through from the new API. But I'd have to do some digging to see if they ever get exposed in a way that the templates can access. The default templates are definitely not using it.

**Comment by robskrob** [comment]
@AmeliaBR thank you for letting me know. I think you are right. 
> The default templates are definitely not using it.

I think [this is](https://github.com/Shopify/buy-button-js/blob/master/src/components/product.js#L218-L221 ) where the data for the carousel image is defined and no `alt text` is set.

**Comment by AmeliaBR** [comment]
Searching this repository, I can't find any reference to the `altText` data field at all. The underlying SDK uses it in the request for image data, but I can't figure out all the layers of abstractions between that response and the data objects that are exposed to the templates.

I'm _pretty_ sure that the relevant connection is the `get image()` method in component/product.js, but there may be other places like the carousel image code that are currently destructuring data and only passing along select named properties.

**Comment by robskrob** [comment]
@AmeliaBR thank you for sharing this [research](https://github.com/Shopify/buy-button-js/issues/369#issuecomment-447491225). I have created a [PR](https://github.com/Shopify/buy-button-js/pull/583) and I have tested my work locally -- I now see the alt text in the PDP images. Hopefully the PR could get reviewed quickly so I can respond to any feedback and so ultimately the PR can get merged. I'd really like to have alt text available on my images.

@AmeliaBR Let me know what you think with regards to the above PR I set up.

**Comment by robskrob** [comment]
@spencercanner I've only interacted with you so far on this project. Would you be able to review [this PR](https://github.com/Shopify/buy-button-js/pull/583)? 

CC: @tessalt, @tanema, @harisaurus

**Comment by AmeliaBR** [comment]
Thanks for doing this @robskrob !  I started to make the same changes myself, but realized it was too complex to do it without testing it out fully.

I left a comment with regards to the image carousel code. I've never used that particular template, so I'd never looked at it closely. The current code is really bad for accessibility. I gave suggestions for a bare-minimum fix and for more complicated options if you are willing to track down all the pieces.
